### PR TITLE
Lots of fixes for Mechs

### PR DESCRIPTION
**Pull request by Sindorman**
Spent lots of time and lots of changes to fix issues:
- bugfix: "Mechs no longer runtime during destroy call."

- bugfix: "Turrets and NPCS no longer target empty mechs."

- bugfix: "Mechs no longer runtime during armor check. Mechs now use their armor and values for it. Before it would ignore them entirely and deal 100% damage."

- Added `ismech()` define and also `QDEL_NULL_LIST()` define from TG. Matt's suggestion

- renamed any `armour` references in mech code to `armor` to keep it consistent

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

File: code/game/machinery/portable_turret.dm [hunk 1]
```diff
@@ -578,6 +578,11 @@
 	if(L.lying)		//if the perp is lying down, it's still a target but a less-important target
 		return lethal ? TURRET_SECONDARY_TARGET : TURRET_NOT_TARGET
 
+	if(ismech(L))
+		var/mob/living/heavy_vehicle/M = L
+		if(!M.pilots || !M.pilots.len)
```
Comment (skull132) [inline code/game/machinery/portable_turret.dm]:
```suggestion
		if(!M.pilots?.len)
```

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -339,6 +339,11 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 //////////////////////////////
 
 /mob/living/simple_animal/hostile/proc/validator_living(var/mob/living/L, var/atom/current)
+	if(ismech(L))
+		var/mob/living/heavy_vehicle/M = L
+		if(!M.pilots || !M.pilots.len)
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]:
```suggestion
		if(!M.pilots?.len)
```

File: code/modules/heavy_vehicle/components/armour.dm [hunk 3]
```diff
@@ -3,24 +3,24 @@
 		damage_flags &= ~(DAM_SHARP | DAM_EDGE)
 	. = ..()*/
 
-/obj/item/robot_parts/robot_component/armour
+/obj/item/robot_parts/robot_component/armor
 	name = "exosuit armour plating"
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]: Change the names too.

File: code/modules/heavy_vehicle/components/armour.dm [hunk 4]
```diff
@@ -3,24 +3,24 @@
...
 	armor = list(melee = 75, bullet = 33, laser = 50, energy = 10, bomb = 25, bio = 100, rad = 0)
 	origin_tech = list(TECH_MATERIAL = 1)
 
-/obj/item/robot_parts/robot_component/armour/radproof
+/obj/item/robot_parts/robot_component/armor/radproof
 	name = "radiation-proof armour plating"
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]: As above.

File: code/modules/heavy_vehicle/components/armour.dm [hunk 5]
```diff
@@ -3,24 +3,24 @@
...
 	desc = "A fully enclosed radiation hardened shell designed to protect the pilot from radiation."
 	armor = list(melee = 75, bullet = 33, laser = 50, energy = 25, bomb = 25, bio = 100, rad = 100)
 	origin_tech = list(TECH_MATERIAL = 3)
 
-/obj/item/robot_parts/robot_component/armour/em
+/obj/item/robot_parts/robot_component/armor/em
 	name = "EM-shielded armour plating"
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]: As above.

File: code/modules/heavy_vehicle/components/body.dm [hunk 6]
```diff
@@ -22,7 +22,7 @@
 /obj/item/mech_component/chassis/update_components()
 	diagnostics = locate() in src
 	cell =        locate() in src
-	armour =      locate() in src
+	mech_armor =      locate() in src
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/body.dm]: Fucked indentation.

File: code/modules/research/designs/mechfab/mechs/designs_exosuits.dm [hunk 7]
```diff
@@ -56,23 +56,23 @@
 /datum/design/item/mechfab/exosuit/radproof_armour
 	name = "radiation-proof exosuit armour"
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/mechfab/mechs/designs_exosuits.dm]: Change this to armor.

File: code/modules/research/designs/mechfab/mechs/designs_exosuits.dm [hunk 8]
```diff
@@ -56,23 +56,23 @@
...
 	name = "EM-shielded exosuit armour"
 	id = "mech_armour_em"
-	build_path = /obj/item/robot_parts/robot_component/armour/em
+	build_path = /obj/item/robot_parts/robot_component/armor/em
 	time = 50
 	req_tech = list(TECH_MATERIAL = 2, TECH_POWER = 2)
 	materials = list(DEFAULT_WALL_MATERIAL = 12500, "silver" = 1000)
 
 /datum/design/item/mechfab/exosuit/combat_armour
 	name = "Combat exosuit armour"
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/mechfab/mechs/designs_exosuits.dm]:
```suggestion
	name = "combat exosuit armor"
```

File: code/modules/research/designs/mechfab/mechs/designs_exosuits.dm [hunk 9]
```diff
@@ -56,23 +56,23 @@
...
 	name = "radiation-proof exosuit armour"
 	id = "mech_armour_radproof"
-	build_path = /obj/item/robot_parts/robot_component/armour/radproof
+	build_path = /obj/item/robot_parts/robot_component/armor/radproof
 	time = 50
 	req_tech = list(TECH_MATERIAL = 2, TECH_ENGINEERING = 2)
 	materials = list(DEFAULT_WALL_MATERIAL = 12500)
 
 /datum/design/item/mechfab/exosuit/em_armour
 	name = "EM-shielded exosuit armour"
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/mechfab/mechs/designs_exosuits.dm]:
```suggestion
	name = "EM-shielded exosuit armor"
```

File: code/modules/research/designs/mechfab/mechs/designs_exosuits.dm (hunk 7 above)
Comment (NonQueueingMatt) [inline code/modules/research/designs/mechfab/mechs/designs_exosuits.dm]:
```suggestion
	name = "radiation-proof exosuit armor"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 3 above)
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "exosuit armor plating"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 4 above)
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "radiation-proof armor plating"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 5 above)
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "EM-shielded armor plating"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 3 above)
Comment (Sindorman) [inline code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "exosuit armor plating"
```

Comment (Sindorman) [reply code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "exosuit armor plating"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 4 above)
Comment (Sindorman) [reply code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "radiation-proof armor plating"
```

File: code/modules/heavy_vehicle/components/armour.dm (hunk 5 above)
Comment (Sindorman) [reply code/modules/heavy_vehicle/components/armour.dm]:
```suggestion
	name = "EM-shielded armor plating"
```

**Comment by KasparoVy** [comment]
If y'all are going that far with the spellchecking, you might as well change the filename from `code/modules/heavy_vehicle/components/armour.dm` to `.../armor.dm` too eh?

**Pull request merged**